openrtb_ext/floors.go

```diff
@@ -0,0 +1,114 @@
+	SkipRate           int                    `json:"skiprate,omitempty"`
+	Location           *PriceFloorEndpoint    `json:"floorendpoint,omitempty"`
+	Data               *PriceFloorData        `json:"data,omitempty"`
+	Enforcement        *PriceFloorEnforcement `json:"enforcement,omitempty"`
+	Enabled            *bool                  `json:"enabled,omitempty"`
+	Skipped            *bool                  `json:"skipped,omitempty"`
+	FloorProvider      string                 `json:"floorprovider,omitempty"`
+	FetchStatus        string                 `json:"fetchstatus,omitempty"`
+	PriceFloorLocation string                 `json:"location,omitempty"`
+}
+
+type PriceFloorEndpoint struct {
+	URL string `json:"url,omitempty"`
+}
+
+type PriceFloorData struct {
+	Currency            string                 `json:"currency,omitempty"`
+	SkipRate            int                    `json:"skiprate,omitempty"`
+	FloorsSchemaVersion string                 `json:"_,omitempty"`
+	ModelTimestamp      int                    `json:"modeltimestamp,omitempty"`
+	ModelGroups         []PriceFloorModelGroup `json:"modelgroups,omitempty"`
+	FloorProvider       string                 `json:"floorprovider,omitempty"`
+}
+
+type PriceFloorModelGroup struct {
+	Currency     string             `json:"currency,omitempty"`
+	ModelWeight  *int               `json:"modelweight,omitempty"`
+	ModelVersion string             `json:"modelversion,omitempty"`
+	SkipRate     int                `json:"skiprate,omitempty"`
+	Schema       PriceFloorSchema   `json:"schema,omitempty"`
+	Values       map[string]float64 `json:"values,omitempty"`
+	Default      float64            `json:"default,omitempty"`
+}
+type PriceFloorSchema struct {
+	Fields    []string `json:"fields,omitempty"`
+	Delimiter string   `json:"delimiter,omitempty"`
+}
+
+type PriceFloorEnforcement struct {
+	EnforceJS     *bool `json:"enforcejs,omitempty"`
+	EnforcePBS    *bool `json:"enforcepbs,omitempty"`
+	FloorDeals    *bool `json:"floordeals,omitempty"`
+	BidAdjustment *bool `json:"bidadjustment,omitempty"`
+	EnforceRate   int   `json:"enforcerate,omitempty"`
+}
+
+type ImpFloorExt struct {
+	FloorRule      string  `json:"floorRule,omitempty"`
+	FloorRuleValue float64 `json:"floorRuleValue,omitempty"`
+	FloorValue     float64 `json:"floorValue,omitempty"`
+}
+type Price struct {
+	FloorMin    float64 `json:"floormin,omitempty"`
+	FloorMinCur string  `json:"floormincur,omitempty"`
+}
+
+type ExtImp struct {
+	Prebid *ImpExtPrebid `json:"prebid,omitempty"`
+}
+
+type ImpExtPrebid struct {
+	Floors Price `json:"floors,omitempty"`
+}
+
+// GetEnabled will check if floors is enabled in request
+func (Floors *PriceFloorRules) GetEnabled() bool {
+	if Floors != nil && Floors.Enabled != nil {
+		return *Floors.Enabled
+	}
+	return true
+}
+
+func (modelGroup *PriceFloorModelGroup) Copy() *PriceFloorModelGroup {
+	newModelGroup := new(PriceFloorModelGroup)
```

Comment:
Since you're accepting a function pointer, please handle the case where `modelGroup` is nil, or perhaps a pointer isn't necessary? In general, it's best to default to not using pointers and make the move if you have a very large data structure or need to keep state, such as with concurrency primitives.

Reply:
Corrected